tests/test_player.py

```diff
@@ -0,0 +1,303 @@
+
+    def test_added_fields(self, players):
+        """
+        Test misc fields that may not be present in historical records
+        """
+        for player in players:
+            if getattr(player, "ritual", None):
+                assert isinstance(player.ritual, str)
+            if getattr(player, "peanut_allergy", None):
+                assert isinstance(player.peanut_allergy, bool)
+            if getattr(player, "fate", None):
+                assert isinstance(player.fate, int)
+
+    def test_simulated_copy(self, players):
+        for player in players:
+            mult = player.simulated_copy(multipliers={"overall_rating": 2.0, "patheticism": 0.01, "cinnamon": 6.9})
+            assert isinstance(mult, Player)
+            assert mult.batting_stars >= player.batting_stars
+            assert mult.patheticism < player.patheticism
+            if getattr(player, "cinnamon", None):
+                assert mult.cinnamon > player.cinnamon
+
+            buff = player.simulated_copy(buffs={"overall_rating": -2.0, "patheticism": -0.99, "cinnamon": 4.20})
+            assert isinstance(buff, Player)
+            assert buff.batting_stars <= player.batting_stars
+            assert buff.patheticism >= player.patheticism
+            if getattr(player, "cinnamon", None):
+                assert buff.cinnamon > player.cinnamon
+
+            reroll = player.simulated_copy(reroll={"overall_rating": True, "patheticism": True, "cinnamon": True})
+            assert isinstance(reroll, Player)
+
+            override = player.simulated_copy(overrides={"cinnamon": 6.9})
+            assert isinstance(override, Player)
+            assert override.cinnamon == 6.9
+
+    def test_stars_bounded(self):
+        """
+        Test that star equations produce correct results
+        """
+        with open(f"{TEST_DATA_DIR}/player/stars.json", "r") as fp:
+            players = [Player(data) for data in json.load(fp)]
+            for player in players:
+                assert player.batting_stars == pytest.approx(player.reference_batting_stars)
+                assert player.pitching_stars == pytest.approx(player.reference_pitching_stars)
+                assert player.baserunning_stars == pytest.approx(player.reference_baserunning_stars)
+                assert player.defense_stars == pytest.approx(player.reference_defense_stars)
+
+    def test_soulscream_bounded(self):
+        """
+        Test that soulscream equation produces correct results
+        """
+        with open(f"{TEST_DATA_DIR}/player/screams.json", "r") as fp:
+            players = [Player(data) for data in json.load(fp)]
+            for player in players:
+                assert player.soulscream == pytest.approx(player.reference_scream)
+
+    def test_vibes_bounded(self):
+        """
+        Test that vibe equation produces correct results
+        """
+        with open(f"{TEST_DATA_DIR}/player/vibes.json", "r") as fp:
```

Comment:
having the expected vibes defined in json makes sense here, but again, I'd consider having multiple test cases rather than a loop. though I'm not even sure if this needs the multiple test cases--the logic is the same regardless of what attribute is high or low, so it's the same amount of coverage for each